pkg/fleet/daemon/daemon.go

```diff
@@ -120,8 +120,9 @@ func (d *daemonImpl) GetAPMInjectionStatus() (status APMInjectionStatus, err err
 	// Host is instrumented if the ld.so.preload file contains the apm injector
 	ldPreloadContent, err := os.ReadFile("/etc/ld.so.preload")
 	if err != nil {
-		return status, fmt.Errorf("could not read /etc/ld.so.preload: %w", err)
+		return status, nil
```

Comment:
Let's just add a case for `os.IsNotExist(err)` instead and set `HostInstrumented = false` if that's the case